### PR TITLE
Fix Assault Vest's battle message and effect on Me First

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Addresses #1872 and #1871.

Source: https://bulbapedia.bulbagarden.net/wiki/Assault_Vest

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Ready for review.

File: src/battle_util.c [hunk 1]
```diff
@@ -1834,7 +1834,7 @@ u8 CheckMoveLimitations(u8 battlerId, u8 unusableMoves, u8 check)
             unusableMoves |= gBitTable[i];
         else if (HOLD_EFFECT_CHOICE(holdEffect) && *choicedMove != 0 && *choicedMove != 0xFFFF && *choicedMove != gBattleMons[battlerId].moves[i])
             unusableMoves |= gBitTable[i];
-        else if (holdEffect == HOLD_EFFECT_ASSAULT_VEST && gBattleMoves[gBattleMons[battlerId].moves[i]].power == 0)
+        else if (holdEffect == HOLD_EFFECT_ASSAULT_VEST && gBattleMoves[gBattleMons[battlerId].moves[i]].power == 0 && gBattleMons[battlerId].moves[i] != MOVE_STUFF_CHEEKS)
```
Comment (ghoulslash) [inline src/battle_util.c:1837]: shouldnt this be Me First?

Comment (AsparagusEduardo) [reply src/battle_util.c]: Forgot to change it ^^;
Ready for re-review.

**Comment by AsparagusEduardo** [comment]
Ready for re-review

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**